File: service-invocation/node/http/frontend/controllers/apiController.js
import { Router } from "express";
import axios from "axios";

const router = Router();

export const apiController = router.get("/", async (_req, res) => {
  try {
    const { data } = await axios.get("http://localhost:3500/v1/invoke/backend/method/api/cars/get");
    res.json({ msg: data });
  } catch (error) {
    console.log(error);
  }
});

File: service-invocation/java/http/frontend/src/main/java/com/dapr/http/Controllers/ApiController.java
package com.dapr.http.Controllers;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

@RestController
public class ApiController {
    @GetMapping("/api/dapr")
    public String apiController() {
        final String uri = "http://localhost:3500/v1.0/invoke/backend/method/api/get/cars";

        RestTemplate restTemplate = new RestTemplate();
        String result = restTemplate.getForObject(uri, String.class);

        System.out.println(result);

        return result;
    }

}
